starlette/datastructures.py

```diff
@@ -13,6 +13,14 @@ class Address(typing.NamedTuple):
     port: int
 
 
+_DefaultType = typing.TypeVar("_DefaultType")
+# Mapping values are covariant, keys are invariant
+_KeyType = typing.TypeVar("_KeyType")  # key type
+_ValueType = typing.TypeVar(
+    "_ValueType", covariant=True
+)  # value type for covariant containers
+
```

Comment:
(Nit) These comments seem a tad redundant with the code and the comment above?

```suggestion
_KeyType = typing.TypeVar("_KeyType")
_ValueType = typing.TypeVar("_ValueType", covariant=True)
```

Plus, any strong opinion against a more concise `_KT, _VT` naming?

I also note that the [`typing` docs](https://docs.python.org/3/library/typing.html) tend to hint covariance in the typevar name everywhere. So, we could use `_KT, _VT_co` to provide an even more consitent UX? _And_ link to the typing docs as a motivation for what we're doing here.

```suggestion
# Follow generics conventions listed here:
# https://docs.python.org/3/library/typing.html#typing.Mapping
# But underscore-prefixed so they don't get exposed as public API.
_KT = typing.TypeVar("_KT")
_VT_co = typing.TypeVar("_VT_co", covariant=True)
```

Reply:
This PR was using `_KT` and `_VT_co` before my comment here: https://github.com/encode/starlette/pull/1449#pullrequestreview-981163983 😅

I thought it will be more friendly to @tomchristie's eyes, but if more members prefer the convention used in the python docs, then I guess it's fine...

Reply:
Ha! I see. :) Didn't read up the full thread. Well, no strong opinions on my side, we can keep things like this.

Reply:
I might still suggest dropping some of the redundant comments:

```suggestion
_KeyType = typing.TypeVar("_KeyType")
_ValueType = typing.TypeVar("_ValueType", covariant=True)
```

Reply:
I think the `_VT_co` comes from the fact that in those modules there's also a `_VT`: values in `Mapping` are covariant, values in `Dict` are invariant.

So in this case, _right now_ we could just use `_ValueType` as we are. But once we add types to `MultiDict` (the subclass) I think we'll need an invariant value type. So I'll rename this to `_CovariantValueType`.

And yes, the comments are now redundant.

Reply:
I landed on:

```python
_KeyType = typing.TypeVar("_KeyType")
# Mapping keys are invariant but their values are covariant since
# you can only read them
# that is, you can't do `Mapping[str, Animal]()["fido"] = Dog()`
_CovariantValueType = typing.TypeVar("_CovariantValueType", covariant=True)
```

Does the example helps illustrate why the values are covariant?